pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,775 @@
+
+    Args:
+        dt (DateTime): the DateTime for which the nanoseconds offset should be returned
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.nanos(dt)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_of_day(dt, tz: TimeZone) -> int:
+    """ Returns a long value of nanoseconds since midnight for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the nanoseconds since midnight
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.nanosOfDay(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_of_second(dt, tz: TimeZone) -> int:
+    """ Returns a long value of nanoseconds since the top of the second for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the nanoseconds
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.nanosOfSecond(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_to_millis(ns: int) -> int:
+    """ Converts nanoseconds to milliseconds.
+
+    Args:
+        ns (int): the value of nanoseconds to convert
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.nanosToMillis(ns)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def nanos_to_datetime(ns: int):
+    """ Converts a value of nanoseconds from Epoch to a DateTime.
+
+    Args:
+        ns (long): the long nanoseconds since Epoch value to convert
+
+    Returns:
+        DateTime
+    """
+    try:
+        return _JDateTimeUtils.nanosToTime(ns)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def plus_period(dt, period):
+    """ Adds a period to a DateTime.
+
+    Args:
+        dt (DateTime): the starting DateTime value
+        period (Period): the Period to add to the DateTime
+
+    Returns:
+        DateTime
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.plus(dt, period)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def plus_nanos(dt, ns: int):
+    """ Adds nanoseconds to a DateTime.
+
+    Args:
+        dt (DateTime): the starting DateTime value
+        ns (int): the long number of nanoseconds to add to DateTime
+
+    Returns:
+        DateTime
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.plus(dt, ns)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def second_of_day(dt, tz: TimeZone) -> int:
+    """ Returns an int value of seconds since midnight for a DateTime in the specified time zone.
```

Comment:
```suggestion
    """ Returns the number of seconds since midnight for a DateTime in the specified time zone.
```
We don't need to say that things return an int, because it is listed below.  We just need to get across what it is doing as clearly as possible. in this sentence.  Check other methods.